### PR TITLE
Dapi 1439 - Migration script for already consolidated axis

### PR DESCRIPTION
**Pull request by AymericPlanche**
**Description (for Contributor and Core Developer)**

Those migration script take into account CE and EE format (one axis vs two axes)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: upgrades/schema/Version_5_0_20201118133700_migrate_product_axis_rate_to_unique_score.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    private ?Connection $db;
+
+    private ?CachedGetLocalesByChannelQuery $getLocalesByChannelQuery;
+
+    public function up(Schema $schema) : void
+    {
+        $this->disableMigrationWarning();
+
+        $this->db = $this->container->get('database_connection');
+        $this->getLocalesByChannelQuery = $this->container->get('Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Persistence\Query\Structure\CachedGetLocalesByChannelQuery');
```
Comment (LaurentPetard) [inline upgrades/schema/Version_5_0_20201118133700_migrate_product_axis_rate_to_unique_score.php:27]: It's painful but we should avoid to use any PIM service, to be sure that the migration will work whatever the future modifications. (The problem has already occurred several times)

**Review by LaurentPetard — APPROVED** [review]
Adding the table `pim_data_quality_insights_product_score` will be done in another migration script?
Why not in this PR?

**Review by rouliane — APPROVED** [review]
GTM once the fix we discussed for the migration is done

**Review by LaurentPetard — APPROVED** [review]

**Comment by lennart380** [comment]
Is this for v5.0 ?
Because i got error: 
```
CRITICAL  [console] Error thrown while running command "pim:product:index --all --env=prod". Message: "An exception occurred while executing 'SELECT latest_score.product_id, latest_score.scores
FROM pim_data_quality_insights_product_score AS latest_score

```
I´m trying to get an upgrade from 4.0 to 5.0 working, I´m on day two now :-/

**Pull request merged**